js/map.js

```diff
@@ -0,0 +1,120 @@
+!(function () {
+  
+  var mapData;
+
+  function jsonSuccess (data) {
+    mapData = JSON.parse(data).features;
+    var map = Map(mapVars);
+    var setMarker = Marker(map);
+
+    for (var i = 0; i < mapData.length; i++) {
+      setMarker(mapData[i]); 
+      console.log(mapData[i]); 
+    }
+  };
+  
+  var mapVars = { // Play with different map attributes here //  
+    style : [
+      {
+        stylers: [
+          { hue: "#00ffe6" },
+          { saturation: -20 }
+        ]
+      },{
+        featureType: "road",
+        elementType: "geometry",
+        stylers: [
+          { lightness: 100 },
+          { visibility: "simplified" }
+        ]
+      },{
+        featureType: "road",
+        elementType: "labels",
+        stylers: [
+          { visibility: "off" }
+        ]
+      }
+    ],
+    canvas : document.getElementById("map-canvas"),
+    centerCoord : { // Nice center point over North America //
+    lat: 39.00, 
+    lng: -100.00 
+    },
+    zoomLevel: 4 //Works well on desktops, may need to adjust/script for mobile friendly
+  };
+
+  var Map = function (mapVars) {
+    var mapCenterPoint = new google.maps.LatLng(
+      mapVars.centerCoord.lat, 
+      mapVars.centerCoord.lng
+    );
+
+    var mapOptions = {
+      center: mapCenterPoint,
+      zoom: mapVars.zoomLevel,
+    }; 
+    
+    var map = new google.maps.Map(mapVars.canvas, mapOptions);
+    map.setOptions({styles: mapVars.style});
+    
+    return map;
+  };
+
+
+  var Marker = function(map) {
+
+    setMarker = function (entry) {
+      var newMarker = marker(entry);
+      addMarkerListener(newMarker, infoWindow(entry));
+      newMarker.setMap(map);
+    };
+
+    addMarkerListener = function (marker, infoWindow) {
+        google.maps.event.addListener(marker, 'click', function() {
+          infoWindow.open(map,marker);
+        });
+    };
+
+    marker = function (entry) {
+      return new google.maps.Marker({
+        position: latLong(entry),
+        map: map,
+        title: entry.properties.pizzeria,
+        animation: google.maps.Animation.DROP
+      });
+    };
+
+    latLong = function (entry) {
+      var coords = entry.geometry.coordinates;
+      return new google.maps.LatLng(coords[1], coords[0]); 
+    };
+
+    infoWindow = function (entry) {
+      var infoWindowContent = contentHtml(entry);
+      return new google.maps.InfoWindow({content: infoWindowContent});
+    };
+
+    contentHtml = function(entry) {
+      var props = entry.properties;
+      return '<div id="content">'+
+        props.pizzeria + 
+        '<br />' +
+        props.address +
+        '<br />' +
+        props.city + 
+        '<br />' +
+        '<a href=' + props.website + '>' + props.website + '</a>'
+      '</div>';
+    };
+    return setMarker;
+  };
+
+
+
+  window.onload = function () {
+  psQuery.ajax({url: 'https://raw.githubusercontent.com/ETNOL/pizza/master/pizza_map.geojson', success: jsonSuccess})
```

Comment:
Either option sounds good! My reasoning for using github to host the json
data was to make it easier to keep updated. I'd probably want to use the
original source repo instead of my fork though so I wouldn't need to pull
down updates. My original map on my personal site wasn't geared towards
anything specific so I might just replace it with this instead. If I do,
should I make a pull request with its url in an examples.md or just drop
the code in a folder like "examples/pizza-map/{codebase}"?

On Fri, Aug 21, 2015 at 3:31 PM, Alan B Smith <notifications@github.com>
wrote:

> In js/map.js
> <https://github.com/stevekinney/pizza/pull/82#discussion_r37680332>:
>
> > +        props.pizzeria +
> > +        '<br />' +
> > +        props.address +
> > +        '<br />' +
> > +        props.city +
> > +        '<br />' +
> > +        '<a href=' + props.website + '>' + props.website + '</a>'
> > +      '</div>';
> > +    };
> > +    return setMarker;
> > +  };
> > +
> > +
> > +
> > +  window.onload = function () {
> > +  psQuery.ajax({url: 'https://raw.githubusercontent.com/ETNOL/pizza/master/pizza_map.geojson', success: jsonSuccess})
>
> This looks great. The only thing I would change is this line. I would set
> the url to './pizza_map.geojson' so anyone could grab from their dataset
> when they pull it down.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/stevekinney/pizza/pull/82/files#r37680332>.
>